CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+
+## Avoid Breaking Changes
+
+The first rule for Azure REST APIs is to avoid breaking changes in GA'd APIs. The [Azure Breaking Changes Policy](aka.ms/AzBreakingChangesPolicy) is a formal description of what changes are considered breaking changes. This rule trumps adherence to Design Guidelines, Coding Style, etc.
+
+## Design Guidelines
+
+Azure REST APIs should conform to the [Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md).
+
+There is a [YouTube video series](https://www.youtube.com/watch?v=9Ng00IlBCtw) by Jeffrey Richter that describes the guidelines in detail.
+
+Another resource is the [Considerations for Service Design](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md), which is an introduction to REST API design mainly for services that are just getting started.
+
+### Exceptions for consistency within a service
+
+There are situations where a service has GA'd their API with design patterns that do not follow our guidelines and it would be a breaking change to adopt the API design in the guidelines.
+Because the first rule is to avoid breaking changes and because we want APIs to be consistent within a service, these design patterns are considered the standard for that service and should be followed in all subsequent (non-breaking) versions of that service's REST API.
+
+## Coding Style
+
+REST APIs for Azure should be defined using the [OpenAPI v2](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) format.
+
+The REST API should also adhere to the conventions documented in the [Azure API Style Guide](https://github.com/Azure/azure-api-style-guide/blob/main/openapi-style-guide.md).
+
+You can use the [Spectral linter](https://meta.stoplight.io/docs/spectral/674b27b261c3c-overview) with our [Spectral ruleset](https://github.com/Azure/azure-api-style-guide#how-to-use-the-spectral-ruleset) to check conformance with OpenAPI v2 and the Style Guide.
+
+## Directory Structure
+
+See the [README.md][./README.md] for a description of the directory structure to use when adding files to this repo.
+
+## Pull Requests
+
+If you want to contribute to the repository, follow these steps:  
+  1. Fork the repository and create a new branch for your changes.
+  2. If you are introducing a new api-version, create a new directory for that api-version and copy all the files from the previous version into the new directory. Make this the very first commit in your branch and then make your changes in subsequent commits.
+  3. Use the [linting tools](#coding-style) to check your changes for compliance with the OpenAPI v2 standard, the Azure REST API Guidelines and that Azure API Style Guide.
+  4. Push to your fork and submit a pull request to the **main** branch using either the "control plane" or "data plane" PR template -- the main PR template lets you select which one.
```

Comment:
This should only be when done. I question how common this is a one-and-done process. I highly recommend people - as I did in our internal docs - to create a feature branch instead e.g., `release-{service}-{RP}-{api-version(-preview)}` they target until they are all done, then submit a PR to main. However, because these feature branches tend to live for a while, they often need to merge main which gets *extremely* messy - hundreds of commits and thousands of unrelated files. Instead, they should rebase, which my instructions cover.

Lots of teams target main like this, and it's wrong because - whether in a preview/ or stable/ directory - they are likely to make breaking changes thinking that, "We haven't released it yet, so it's okay to change what's in main." According to past conversations with @JeffreyRichter and @johanste this is incorrect. Being in main means published - effectively immutable.

Reply:
```suggestion
  4. Push to changes to the branch in your fork until the changes are ready to be integrated.
  5. Rebase your branch if needed to incorporate any changes to **main** and submit a pull request to the **main** branch using either the "control plane" or "data plane" PR template -- the main PR template lets you select which one.
```